### PR TITLE
aws-default-cp-instance-type

### PR DESCRIPTION
**Pull request by calvix**
add defaulting for instanceType used for control plane machine
(it was empty before and cluster creation failed)

### CODE REVIEW

File: helm/policies-aws/templates/AWSCAPI.yaml [hunk 1]
```diff
@@ -113,3 +113,20 @@ spec:
               {{ `{{` }} regex_replace_all('^$', '{{ `{{` }}@{{ `}}` }}', '{{ .Values.awsClusterRoleIdentity.sourceIdentityRef.name}}') {{ `}}` }}
             kind: |-
               {{ `{{` }} regex_replace_all('^$', '{{ `{{` }}@{{ `}}` }}', '{{ .Values.awsClusterRoleIdentity.sourceIdentityRef.kind }}') {{ `}}` }}
+  # awsmachinetemplate-default-empty-string adds default values to fields in a AWSMachineTemplate CR if it has the watch-filter label and the values are empty strings.
+  # its used to default instanceType for a control plane machine in case its empty
+  - name: awsmachinetemplate-default-empty-string
```
Comment (MarcelMue) [inline helm/policies-aws/templates/AWSCAPI.yaml:118]: Do you only need this for empty strings? Or also if the instanceType is completely missing?

Comment (calvix) [reply helm/policies-aws/templates/AWSCAPI.yaml:118]: what is the differnece? hor how should I do that? i assumed empty string is same as missing

File: helm/policies-aws/templates/AWSCAPI.yaml [hunk 2]
```diff
@@ -113,3 +113,35 @@ spec:
               {{ `{{` }} regex_replace_all('^$', '{{ `{{` }}@{{ `}}` }}', '{{ .Values.awsClusterRoleIdentity.sourceIdentityRef.name}}') {{ `}}` }}
             kind: |-
               {{ `{{` }} regex_replace_all('^$', '{{ `{{` }}@{{ `}}` }}', '{{ .Values.awsClusterRoleIdentity.sourceIdentityRef.kind }}') {{ `}}` }}
+  # awsmachinetemplate-default adds default values to fields in a AWSMachineTemplate CR if it has the watch-filter label and the values are completely missing.
+  - name: awsmachinetemplate-default
+    match:
```
Comment (calvix) [inline helm/policies-aws/templates/AWSCAPI.yaml:118]: is this the right way?

**Review by MarcelMue — APPROVED** [review]
LGTM

**Pull request merged**